### ISSUE TITLE
[FromQuery] does not bind arrays unless Name specified

### ISSUE THREAD

**Issue by yogster** [aspnet/Mvc#8034]
This is either a bug or I am missing something about how binding attributes work.

In the following controller action:

    [Route("api/[controller]")]
    [ApiController]
    public class ValuesController : ControllerBase
    {
        [HttpGet]
        public ActionResult<int> Get([FromQuery]int[] numbers)
        {
            return numbers.Sum();
        }
    }

When requesting the URL: `http://localhost:49965/api/values?numbers=1&numbers=2`, `numbers` will always be an empty array and the action will always return zero. The same happens with `IEnumerable<int>`

If the `Name` property is specified in `FromQuery`, model binding works as expected:

    [HttpGet]
    public ActionResult<int> Get([FromQuery(Name = "numbers")]int[] numbers)
    {
        return numbers.Sum(); //The URL above returns 3, as expected.
    }

However if the parameter is changed to `int`, I don't need to specify `Name` explicitly:

    [HttpGet]
    public ActionResult<int> Get([FromQuery]int numbers)
    {
        return numbers; //The URL above returns 1, as expected.
    }

I would expect the same behaviour (using the parameter name by default) when binding to arrays or other collections.

Using `Microsoft.AspNetCore.App 2.1.1`

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @yogster.
@dougbu, can you please look into this? Thanks!

**Comment by dougbu** [comment]
@yogster thank you for your feedback.

We're closing this issue as we think this is fixed in our working code base. In particular, this report matches #7712. Would you be able to try our nightly builds and see if the issue is resolved for you?

Nightly builds are available from https://dotnet.myget.org/gallery/aspnetcore-dev.

**Comment by yogster** [comment]
Thanks for your prompt response, @dougbu. Is this fixed in 2.2.0 or in 2.1.3?

I've tried using Microsoft.AspNetCore.App v2.1.3-rtn-30898 but my test app exits immediately with `﻿﻿The program '[1856] dotnet.exe' has exited with code -2147450730 (0x80008096).`. I appreciate your effort and I'm happy to test, but I need you to point me in the right direction.

**Comment by dougbu** [comment]
`0x80008096` corresponds to a `FrameworkSetupError`. Likely means you don't have the correct version of `dotnet` to use the latest Microsoft.AspNetCore.App package. Suggest installing the latest SDK for your platform from https://github.com/dotnet/core-sdk#installers-and-binaries.

**Comment by CrispyDrone** [comment]
Hi,

I have a controller method where I accept an IEnumerable of an enum (and some additional properties):
```csharp
public IActionResult GetAllLookups([FromQuery] IEnumerable<lookupEnum> lookups, Languages language = Languages.NL)
        {
            var result = _candidateService.GetAllLookupEnts(PermissionsUser.Write, language, lookups);
            return Ok(result);
        }
```
When leaving out `[FromQuery]` and passing this in the body of the request, the binding works without issue.

However when using `[FromQuery]`, it only works when specifying the Name property like so:
`[FromQuery(Name = "lookups")]`

**Comment by dougbu** [comment]
@CrispyDrone this bug (really #7712) was fixed in 2.1.5 and 2.2.0-preview1 or later. Which version of Microsoft.AspNetCore.Mvc are you testing?
